pyproject.toml

```diff
@@ -0,0 +1,76 @@
+[project]
+classifiers=[
+  "Operating System :: OS Independent",
+  "Intended Audience :: Science/Research",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  'Operating System :: Microsoft :: Windows',
+  'Operating System :: Unix',
+  'Operating System :: MacOS',
+  "Topic :: Scientific/Engineering",
+]
+
+dynamic = ["version"]
+keywords = ["unstructured grids"]
+description = "Xarray-styled package for reading and directly operating on unstructured grid datasets following UGRID conventions"
+license = {text = "Apache-2.0"}
+name = "uxarray"
+readme = "README.md"
+requires-python = ">=3.9"
+
+# minimal dependencies start
+dependencies = [
+  "xarray",
+  "cartopy",
+  "dask",
+  "numba",
+  "numpy",
+  "scikit-learn",
+  "scipy",
+  "pyarrow",
+  "pandas<2.1.0",
+  "netcdf4",
+  "requests"
+]
+# minimal dependencies end
+
+[project.optional-dependencies]
+complete = ["uxarray[viz, math, dev]"]
+viz = [
+  'datashader',
+  'holoviews',
+  'cartopy',
+  'geoviews',
+  'antimeridian',
+  'matplotlib-base',
+  'matplotlib-inline',
+  'shapely',
+  'spatialpandas'
+]
+math = ['gmpy2', 'pyfma']
+dev = ['pathlib', 'pre_commit', 'pytest', 'pytest-cov']
+
+[project.urls]
+Documentation = "uxarray.readthedocs.io/"
+Source = "https://github.com/UXARRAY/uxarray"
+Tracker = "https://github.com/UXARRAY/uxarray/issues"
+
+
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "setuptools>=60",
+  "setuptools-scm>=8",
+]
+
+[tool.setuptools]
+packages = ["uxarray"]
+
+[tool.setuptools_scm]
+fallback_version = "9999"
```

Comment:
This line as well as the lines from 62 to 67 guarantees us to be safe even with teh removal of `setup.py`, right? Asking this because, for instance the xarray repo has a `setup.py` file with a minimal set of commands that seem to be doing the same task as here

Reply:
![image](https://github.com/UXARRAY/uxarray/assets/67855069/6f105e84-707b-46cb-8d85-836dc393b584)

In most cases we wouldn't run into any issues, however for combability this is what is mentioned in the documentation. I don't think it hurts to keep this included.